packages/create-cloudflare/src/common.ts

```diff
@@ -197,3 +200,82 @@ export const printSummary = async (ctx: PagesGeneratorContext) => {
 	}
 	endSection("See you again soon!");
 };
+
+export const offerGit = async (ctx: PagesGeneratorContext) => {
+	const insideGitRepo = await isInsideGitRepo(ctx.project.path);
+
+	if (insideGitRepo) return;
+
+	ctx.args.git ??= await confirmInput({
+		question: "Do you want to use git?",
+		renderSubmitted: (value: boolean) => `${brandColor(value ? `yes` : `no`)}`,
+		defaultValue: true,
+	});
+
+	if (ctx.args.git) {
+		await printAsyncStatus({
+			promise: initializeGit(ctx.project.path),
+			startText: "Initializing git",
+			doneText: `${brandColor("initialized")} ${dim(`git`)}`,
+		});
+	}
+};
+
+export const gitCommit = async (ctx: PagesGeneratorContext) => {
+	const insideGitRepo = await isInsideGitRepo(ctx.project.path);
+
+	if (!insideGitRepo) return;
+
+	await printAsyncStatus({
+		async promise() {
+			await execa("git", ["add", "."]);
+			await execa("git", [
+				"commit",
+				"-m",
+				"Initial commit (by Create-Cloudflare CLI)",
+			]);
+		},
+		startText: "Committing new files",
+		doneText: `${brandColor("git")} ${dim(`initial commit`)}`,
+	});
+};
+
+/**
+ * Check whether the given current working directory is within a git repository
+ * by looking for a `.git` directory in this or an ancestor directory.
+ */
+export async function isInsideGitRepo(cwd: string) {
+	const res = await findUp(".git", { cwd, type: "directory" });
+	return res !== undefined;
+}
+
+/**
+ * Initialize a new Worker project with a git repository.
+ *
+ * We want the branch to be called `main` but earlier versions of git do not support `--initial-branch`.
+ * If that is the case then we just fallback to the default initial branch name.
+ */
+export async function initializeGit(cwd: string) {
+	try {
+		// Get the default init branch name
+		const { stdout: defaultBranchName } = await execa("git", [
+			"config",
+			"--get",
+			"init.defaultBranch",
+		]);
+
+		// Try to create the repository with the HEAD branch of defaultBranchName ?? `main`.
+		await execa(
+			"git",
```

Comment:
What happens if `git` is not installed?

Reply:
The process would exit with `commmand not found` printed. (The promise would `reject` with `new Error(output, { cause: code })` and the top-level handler would catch it and call `crash` with its `message`.)

Reply:
Now we're using runCommand everywhere, I can confirm that if a command fails for whatever reason it (output = stdout + stderr) will be printed to the console and the process will exit with 1.

Did you have a better way to handle git not being installed?

Reply:
I was thinking we would test for git and if not installed we just wouldn't even ask the question of whether to use git and therefore not even try to create a commit...